cloud/pkg/cloudhub/dispatcher/message_dispatcher.go

```diff
@@ -265,6 +265,17 @@ func (md *messageDispatcher) enqueueAckMessage(nodeID string, msg *beehivemodel.
 		return
 	}
 
+	klog.Errorf("MessageRoute.Resource: %s", msg.GetResource())
+	resourceType, _ := messagelayer.GetResourceType(*msg)
+	if resourceType == beehivemodel.ResourceTypeNamespace {
+		resourceNamespace = resourceName
+	}
+	klog.Errorf("resourceNamespace original: %s", resourceNamespace)
+	if resourceType != beehivemodel.ResourceTypeNode && resourceNamespace == "null" {
+		resourceNamespace = "default"
```

Comment:
I think this does fall back to `default` namespace if that is not set in the message to downstream?

this is follow-up bugfix to address https://github.com/kubeedge/kubeedge/issues/4819#issuecomment-1628392110?

Reply:
that sounds promising to me, can you create the new issue to KubeEdge for this as follow-up of https://github.com/kubeedge/kubeedge/issues/4819?

Reply:
https://github.com/kubeedge/kubeedge/issues/5042 I'd like to discuss the solution with KubeEdge teams first, I don't have much confidence with my current patch...

Reply:
Okay this requires updating KubeEdge v1.14 or later, and that means we need to use Kubernetes v1.24, that deprecated docker-shim. as open source perspective, this is resolved, i will go ahead to close this comment.